definitions/features/foreman_proxy.rb

```diff
@@ -3,19 +3,63 @@ class Features::ForemanProxy < ForemanMaintain::Feature
     label :foreman_proxy
   end
 
-  attr_reader :dhcpd_conf_file, :cert_path
+  attr_reader :dhcpd_conf_file, :cert_path, :request_url,
+              :default_ssl_cert_options
 
   def initialize
     @dhcpd_conf_file = '/etc/dhcp/dhcpd.conf'
     @cert_path = ForemanMaintain.config.foreman_proxy_cert_path
+    @request_url = "https://#{hostname}:9090/dhcp"
+    @default_ssl_cert_options = load_default_ssl_cert_options
   end
 
   def valid_dhcp_configs?
     dhcp_req_pass? && !syntax_error_exists?
   end
 
+  def list_reservations(subnet)
+    resp = dhcp_restclient["/#{subnet}"].get { |response, _request, _result| response }
+    handle_api_response(resp, true)
+  end
+
+  def add_reservation(subnet, ip, mac_address, name)
+    payload = { :ip => ip, :mac => mac_address, :name => name }
+    resp = dhcp_restclient["/#{subnet}"].post(
+      payload, :content_type => :json, :accept => :json
+    ) { |response, _request, _result| response }
+    handle_api_response(resp)
+  end
+
+  def remove_reservation(subnet, ip)
+    delete_api = api_deprecated_for_delete? ? "/#{subnet}/ip/#{ip}" : "/#{subnet}/#{ip}"
+    resp = dhcp_restclient[delete_api].delete { |response, _request, _result| response }
+    handle_api_response(resp)
+  end
+
   private
 
+  def load_default_ssl_cert_options
```

Comment:
I would prefer an SSL utility class that accepts following file paths as default arguments.
And `SSL#to_h` would return the desired hash returned here.
Sounds better?